libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -309,12 +304,9 @@ void AP_Terrain::update(void)
     // just schedule any needed disk IO
     schedule_disk_io();
 
-    // try to ensure the home location is populated
-    float height;
-    height_amsl(ahrs.get_home(), height, false);
```

Comment:
I considered it unneeded as it's just caching the home terrain altitude, it will be loaded already as the current position is already polled routinely. The only need to do this is to store it as a shortcut, but again because that terrain area is already loaded I'm not to concerned about that. I could see the argument that if you aren't flying (and aren't logging) and set home to be a remote location that you could have a concern. But that is really edge case. We can leave it in, it just seemed to me like this was an unneeded call that is safely managed for that.